### ISSUE TITLE
Decoders using `RidgeCV`/`RidgeClassifierCV` may not always be taking advantage of built-in CV capabilities

### ISSUE THREAD

**Issue by michellewang** [nilearn/nilearn#3599]
Some Decoder objects can use underlying `sklearn` models with built-in cross-validation (e.g., [`RidgeCV`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.RidgeCV.html) and [`RidgeClassifierCV`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.RidgeClassifierCV.html)). In those cases the `param_grid` input and `cv_params_` fitted attribute are handled in the same way as for the models without built-in CV, which leads to potentially suboptimal/unintuitive behaviour:
1. If a typical dict of sequences of hyperparameter values (e.g., `{'alphas': [0.1, 1, 10]}`) is passed to `param_grid`, then the Decoder will fit one `RidgeCV`/`RidgeClassifierCV` model per hyperparameter value, which means that the model's built-in CV will not be used. Instead, to take advantage of the built-in CV, the user has to wrap the sequence of hyperparameter values in another sequence (see code snippet below), but that is not intuitive and is inconsistent with the models without built-in CV.
2. If the user does pass in a `param_grid` like `{'alphas': [[0.1, 1, 10]]}` as described above, the `cv_params_` attribute of the fitted Decoder will show the full sequence (`[0.1, 1, 10]` here) for each CV iteration (see example below), which is unexpected because there should only be one value per outer CV iteration (per hyperparameter).
3. `cv_params_` will be empty if `param_grid` is not given, even though the model still does CV on the default `alphas` sequence.

Here is some code (adapted from [here](https://github.com/nilearn/nilearn/issues/2886#issuecomment-868760488)) illustrating some of my points above:
```python
from nilearn import datasets
from sklearn.model_selection import train_test_split
from nilearn.decoding import DecoderRegressor

n_subjects = 10 
oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=n_subjects)
gray_matter_map_filenames = oasis_dataset.gray_matter_maps
age = oasis_dataset.ext_vars['age'].astype(float)
gm_imgs_train, gm_imgs_test, age_train, age_test = train_test_split(
    gray_matter_map_filenames, age, train_size=.6, random_state=0)

decoder = DecoderRegressor(
    estimator='ridge', 
    param_grid = {
        'alphas':[[0.1, 1, 10]], # list of list
    },
    cv=2,
)

decoder.fit(gm_imgs_train, age_train)
print(decoder.cv_params_) # {'beta': {'alphas': [[0.1, 1, 10], [0.1, 1, 10]]}}
```

I'd like to make a PR to address these points. My proposed changes for the case where the model is `RidgeCV` or `RidgeClassifierCV` are:
1. If `param_grid` is a sequence of values, wrap it in a list.
2. Within the [inner CV loop](https://github.com/nilearn/nilearn/blob/c33cca7a/nilearn/decoding/decoder.py#L220) (which iterates over hyperparameter values), set `best_param` to the single best value for each hyperparameter instead of the full list. This value should be available from the fitted estimator (e.g., in the `alpha_` attribute for `RidgeCV`/`RidgeClassifierCV`).
3. @jeromedockes also suggested we overwrite the default `alphas` value with a large list of 10-20 log-spaced values, so in most cases the user would not have to specify their own `param_grid`.

### Benefits to the change
1. Users would be able to benefit from the built-in CV capabilities of the models even if they specify `param_grid` as they would for other models.
2. The `cv_params_` attribute of the fitted Decoder would be more informative.

**Comment by ymzayek** [comment]
Sounds great @michellewang!

**Comment by michellewang** [comment]
Thanks @jeromedockes! 

> (ii) in a second step we might want to take advantage of the built-in CV capabilities of other estimators such as logisticregression, lasso

Sounds good, this is also mentioned in #2200. Should this be in another PR or can it be the same one?

**Comment by jeromedockes** [comment]
> Sounds good, this is also mentioned in #2200. Should this be in another PR or can it be the same one?

I would recommend doing it in another PR; this way the first PR will be
smaller and merged sooner. I just mentioned it here so that when you
make the changes for RidgeCV, you have in the back of your mind that we
may want to generalize this to other CV-able estimators too later on.
